presto-tests/src/test/java/com/facebook/presto/tests/TestTpchDistributedStats.java

```diff
@@ -0,0 +1,233 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.tests;
+
+import com.facebook.presto.Session;
+import com.facebook.presto.cost.PlanNodeCost;
+import com.facebook.presto.execution.QueryPlan;
+import com.facebook.presto.execution.StageInfo;
+import com.facebook.presto.spi.QueryId;
+import com.facebook.presto.spi.statistics.Estimate;
+import com.facebook.presto.sql.planner.optimizations.PlanNodeSearcher;
+import com.facebook.presto.sql.planner.plan.PlanNode;
+import com.facebook.presto.sql.planner.plan.PlanNodeId;
+import com.facebook.presto.sql.planner.planPrinter.PlanNodeStats;
+import com.facebook.presto.sql.planner.planPrinter.PlanNodeStatsSummarizer;
+import com.facebook.presto.tpch.ColumnNaming;
+import com.google.common.base.Throwables;
+import com.google.common.collect.ImmutableMap;
+import com.google.common.io.Resources;
+import org.testng.annotations.Test;
+
+import java.io.IOException;
+import java.nio.charset.Charset;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.function.BinaryOperator;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+import java.util.stream.Stream;
+
+import static com.facebook.presto.execution.StageInfo.getAllStages;
+import static com.facebook.presto.tests.TestTpchDistributedStats.MetricComparison.Result.DIFFER;
+import static com.facebook.presto.tests.TestTpchDistributedStats.MetricComparison.Result.MATCH;
+import static com.facebook.presto.tests.TestTpchDistributedStats.MetricComparison.Result.NO_BASELINE;
+import static com.facebook.presto.tests.TestTpchDistributedStats.MetricComparison.Result.NO_ESTIMATE;
+import static com.facebook.presto.tests.tpch.TpchQueryRunner.createQueryRunnerWithoutCatalogs;
+import static com.facebook.presto.util.MoreMaps.mergeMaps;
+import static com.google.common.collect.Maps.transformValues;
+import static java.lang.String.format;
+import static java.util.Arrays.asList;
+import static java.util.Collections.emptyList;
+import static java.util.Collections.emptyMap;
+import static java.util.stream.Collectors.groupingBy;
+import static java.util.stream.Collectors.toMap;
+
+public class TestTpchDistributedStats
```

Comment:
Add some javadoc what it does and what does it output (e.g: it never fails because of assertions)